File: Day 1/1b/descending_window.cpp
#include<iostream>
#include<fstream>
#include<vector>

#include<numeric>

#define BOT -1

using namespace std;

int readthree (ifstream& input, int& v) {
    string line;
    vector<int> V;
    for (int i = 0; i < 3 && getline(input, line); ++i)
        V.push_back(stoi(line));
    v = accumulate(begin(V), end(V), 0);
    return V.size() == 3;
}

int main() {
    int n, m;
    ifstream input ("input");
    vector<int> V;
    string line;
    while (getline(input, line))
        V.push_back(stoi(line));
    
    int previous = V[0] + V[1] + V[2];
    int actual;
    int result = 0;
    for (int i = 3; i < V.size(); ++i) {
        actual = previous + V[i] - V[i-3];
        result += actual > previous;
        previous = actual;
    }
    
    cout << result << endl;

    ofstream out;
    out.open ("output");
    out << result;
    out.close();
    return 0;
}

File: Day 1/1b/descending_window.py
M = [line.rstrip() for line in open('input')]
D = [[M[i-2], M[i-1], M[i]] for i in range(5, len(M), 3) if 
      M[i-2]+ M[i-1]+ M[i] > M[i-5]+ M[i-4]+ M[i-3]
]

output = open('output', 'w')
output.write(str(len(D)))
output.close()